rewrite-java/src/main/java/org/openrewrite/java/tree/MethodTypeBuilder.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2020 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.tree;
+
+import com.koloboke.collect.set.hash.HashObjSet;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class MethodTypeBuilder {
```

Comment:
I am still experimenting with a builder to make method type construction a little easier. I am open to any and all suggestions on where something like this should live?

Reply:
Having it alongside `TreeBuilder` seems reasonable to me.

Reply:
I'm sure at some point a "take this code and generate a tree out of it" interface that is language neutral.